api/api.go

```diff
@@ -11,15 +16,20 @@ type API struct {
 	Router *mux.Router
 }
 
-func Setup(ctx context.Context, r *mux.Router) *API {
+type AuthHandler interface {
+	Require(required auth.Permissions, handler http.HandlerFunc) http.HandlerFunc
+}
+
+func Setup(ctx context.Context, r *mux.Router, permissions AuthHandler) *API {
 	api := &API{
 		Router: r,
 	}
 
 	nopSess := &NOPSessions{}
 	nopCache := &NOPCache{}
+	create := auth.Permissions{Create: true}
 
-	r.HandleFunc("/session", CreateSessionHandlerFunc(nopSess, nopCache)).Methods("POST")
+	r.HandleFunc("/session", permissions.Require(create, CreateSessionHandlerFunc(nopSess, nopCache))).Methods("POST")
```

Comment:
The naming of the endpoint should be plural so `/sessions` as we will be dealing with multiple sessions ideally and it is the conventional way of naming it

Reply:
You are right, standard does suggest to pluralise the endpoints so have changed it.